### PR TITLE
Wide-scale structure rewrite and cleanup.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Opening for Travis, DNM.
- [X] Spend like six hours refactoring the structure material and smoothing stuff to be consistent and centralized. Jesus wept. 
- [X] Remove all the lists that are inexplicably defined on every structure in the game.
- [x] Fix material grilles and girders.
- [x] Fix missing icons with `NORTH`-facing chairs.
- [x] Consider generalizing the icon smoothing crap windows/grilles/low walls use.
- [x] Backport the Europa grille and girder icons, and whitescale them.
- [x] Get the map to pass Travis.

Girder icons were done for Europa by @SkowronX, I have just whitened them and made them crappier in the process.

### CODE REVIEW

File: code/game/objects/structures.dm [hunk 1]
```diff
@@ -32,13 +87,7 @@
...
 
 /obj/structure/Destroy()
-	reset_mobs_offset()
-	var/turf/T = get_turf(src)
-	if(T && parts)
-		new parts(T)
-	. = ..()
-	if(istype(T))
-		T.fluid_update()
+	. = dismantle(do_not_destroy = TRUE) && ..()
```
Comment (afterthought2) [inline code/game/objects/structures.dm]: Seems like `dismatle` returns a bool, not a qdel hint.

File: code/game/objects/structures.dm [hunk 2]
```diff
@@ -160,27 +212,41 @@
...
+				for(var/b_type in GLOB.default_blend_objects)
 					if(istype(O, b_type))
 						success = 1
 						for(var/obj/structure/S in T)
 							if(istype(S, src))
 								success = 0
-						for(var/nb_type in noblend_objects)
+						for(var/nb_type in GLOB.default_noblend_objects)
 							if(istype(O, nb_type))
 								success = 0
```
Comment (afterthought2) [inline code/game/objects/structures.dm]: Suggest writing a helper for the above:
```
/obj/structure/proc/can_blend_with(atom/A)
	if(is_type_in_list(A, GLOB.default_noblend_objects))
		return FALSE
	if(istype(A, type)
		return FALSE // Why does it make this check?
	if(is_type_in_list(A, GLOB.default_blend_objects))
		return TRUE
	return FALSE
```
One of the worst things about this code currently is that the above logic is incomprehensible and copy-pasted in several places.
It's not obvious to me why the blend type lists should be global.

File: code/game/objects/structures/grille.dm [hunk 3]
```diff
@@ -1,3 +1,6 @@
+GLOBAL_LIST_INIT(grille_blend_objects,   list(/obj/machinery/door, /turf/simulated/wall))
+GLOBAL_LIST_INIT(grille_noblend_objects, list(/obj/machinery/door/window))
```
Comment (afterthought2) [inline code/game/objects/structures/grille.dm]: Where are these used?

File: code/game/objects/structures/lattice.dm [hunk 4]
```diff
@@ -8,33 +8,27 @@
...
+	if(!istype(src.loc, /turf/space) && !istype(src.loc, /turf/simulated/open))
+		return INITIALIZE_HINT_QDEL
 	for(var/obj/structure/lattice/LAT in loc)
 		if(LAT != src)
 			crash_with("Found multiple lattices at '[log_info_line(loc)]'")
 			qdel(LAT)
-	update_icon()
-	if(!mapload)
-		update_neighbors()
+	update_neighbors()
```
Comment (afterthought2) [inline code/game/objects/structures/lattice.dm]: I believe these calls are generally not safe until `LateInitialize`, as they can make further update calls (icons, notably) which require `Initialize` to have run to not runtime (possibly due to materials setup). That's roughly why you see all the LateInitialize everywhere; it used to do New/Initialize abuse or spawns to deal with the same issue.

File: code/game/objects/structures/grille.dm (hunk 3 above)
Comment (MistakeNot4892) [reply code/game/objects/structures/grille.dm]: Ah they aren't, I was breaking them out into globals before realizing they all use the exact same list.

File: code/game/objects/structures/lattice.dm (hunk 4 above)
Comment (MistakeNot4892) [reply code/game/objects/structures/lattice.dm]: Implemented LateInitialize, not sure why this isn't marked as outdated.

File: code/game/objects/structures.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/game/objects/structures.dm]: Moved this into a helper, but not quite as granular as this one.

**Comment by MistakeNot4892** [comment]
@afterthought2 would you mind taking another look over? I think I got most of the problem areas (though I haven't touched turfs yet).

File: code/game/objects/structures.dm [hunk 5]
```diff
@@ -1,25 +1,80 @@
...
+		material = _mat
+	if(ispath(material, /material))
+		material = SSmaterials.get_material_datum(material)
+	if(ispath(_reinf_mat, /material))
+		reinf_material = _reinf_mat
+	if(ispath(reinf_material, /material))
+		reinf_material = SSmaterials.get_material_datum(reinf_material)
+	. = ..()
+	if(material || reinf_material)
+		update_materials()
```
Comment (afterthought2) [inline code/game/objects/structures.dm]: It's possible this is unsafe until `LateInitialize`; things used to try and update connections from here, which would then check the neighbors' materials, which would not have run the code just above, etc.

File: code/game/objects/structures/window.dm [hunk 6]
```diff
@@ -21,52 +23,39 @@
...
+		if(is_fulltile())
+			layer = FULL_WINDOW_LAYER
+		. = INITIALIZE_HINT_LATELOAD
 
 // Updating connections may depend on material properties.
-/obj/structure/window/LateInitialize(mapload, start_dir=null, constructed=0, var/new_material, var/new_reinf_material)
+/obj/structure/window/LateInitialize()
 	..()
-	set_anchored(!constructed) // calls update_connections, potentially
+	//set_anchored(!constructed) // calls update_connections, potentially
```
Comment (afterthought2) [inline code/game/objects/structures/window.dm]: Remove or readd, depending on if the logic is needed?

**Comment by MistakeNot4892** [comment]
Had to ditch a few hours of work on this branch due to bugs I couldn't chase down. Sorry @afterthought2, I appreciate the review especially as I requested it, but I am going to get this working and merged before looking into further PRs to clean up the other issues. The smoothing system is a bit of a trainwreck with a ton of copypasted code that needs condensing, but so far if I touch walls or machinery it all falls over and bursts into flames.

**Pull request merged**